### PR TITLE
DOC add example to reorder levels

### LINKED ISSUE #42124: DOC add example to reorder levels
https://pandas.pydata.org/pandas-docs/dev/reference/api/pandas.DataFrame.reorder_levels.html?highlight=reorder_levels#pandas.DataFrame.reorder_levels

No examples! Can we add at least one?

### PR DESCRIPTION
**Pull request by trevorkask**
Response to changes suggested on my previous pull request

- [x] closes #42124
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Have you tried building the docs to check this looks right? 

https://pandas.pydata.org/pandas-docs/stable/development/contributing_documentation.html

**Comment by attack68** [comment]
Are you going to close #42667 or this? You should really continue that PR and close this one.

File: pandas/core/frame.py [hunk 1]
```diff
@@ -6790,7 +6790,33 @@ def swaplevel(self, i: Axis = -2, j: Axis = -1, axis: Axis = 0) -> DataFrame:
             assert isinstance(result.columns, MultiIndex)
             result.columns = result.columns.swaplevel(i, j)
         return result
-
+        """
+        Examples
+        --------
```
Comment (MarcoGorelli) [inline pandas/core/frame.py:6795]: This won't render, it needs to be inside the docstring

File: pandas/core/frame.py [hunk 2]
```diff
@@ -6790,7 +6790,33 @@ def swaplevel(self, i: Axis = -2, j: Axis = -1, axis: Axis = 0) -> DataFrame:
...
+        >>> data={'class':['Mammals','Mammals','Reptiles','Reptiles'
+        ,'Amphibians','Amphibians'],
+         'diet':['Omnivore','Carnivore','Carnivore','Herbivore','Omnivore','Herbivore'],
+         'species':['Humans','Dogs','Snakes','Iguanas','Frogs','Aquatic Salamanders']
+         }
+        >>> df=pd.DataFrame(data,columns=['class','diet','species'])
+        >>> df.set_index(['class','diet'],drop=True,inplace=True)
```
Comment (MarcoGorelli) [inline pandas/core/frame.py:6802]: can you write this without using `inplace=True` please?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Can you try building the docs for this function locally to check it renders fine? (hint: at the moment it won't, this'll help you tell whether the PR is ready)

**Comment by jreback** [comment]
use this one

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Hi @trevorkask - the example looks good, there's just some stylistic errors

Can you fix them up? See https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html?highlight=pre%20commit#pre-commit

**Review by MarcoGorelli — APPROVED** [review]

**Comment by MarcoGorelli** [comment]
Thanks @trevorkask

**Pull request merged**